packages/inquirer/lib/utils/paginator.js

```diff
@@ -7,20 +7,18 @@ var _ = {
 var chalk = require('chalk');
 
 /**
- * The paginator keeps track of a pointer index in a list and returns
- * a subset of the choices if the list is too long.
+ * The paginator returns a subset of the choices if the list is too long.
  */
 
 class Paginator {
-  constructor(screen) {
-    this.pointer = 0;
+  constructor(screen, isInfinite) {
```

Comment:
Let's pass this as option `{ isInfinite }`. Just a boolean argument won't convey the meaning.

Reply:
Also, I believe all major versions on node supports defaults parameter `{ isInfinite = true }` and we can remove the type check default later in the code.